### PR TITLE
Fix 1174

### PR DESCRIPTION
**Pull request by AlexVild**
Fixes issues with guild banks where too many slots were being displayed. Adds a new "MaxBankSlots" param in options under BankOptions that allows the dev to configure their absolute maximum bank slots.

Users should definitely:

- Back up their databases
- Run the server, then close it before migrating. This will re-validate their config, which will sort out the name changes and new options.
- Review the server config's bank slots (Player max slots, guild max slots, max possible slots) values before running the migration.
- After confirming their new config changes, run the migration

### CODE REVIEW

**Review by lodicolo — CHANGES_REQUESTED** [review]

File: Intersect (Core)/Config/Options.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public class Options
 
         public DatabaseOptions PlayerDatabase = new DatabaseOptions();
 
-        [JsonProperty("Player")] public PlayerOptions PlayerOpts = new PlayerOptions();
+        [JsonProperty("Player")] public PlayerOptions PlayerOpts = new  PlayerOptions ();
```
Comment (Arufonsu) [inline Intersect (Core)/Config/Options.cs:55]: unnecessary extra white spacing

**Review by Arufonsu — CHANGES_REQUESTED** [review]

File: Intersect (Core)/Config/Options.cs (hunk 1 above)
Comment (AlexVild) [reply Intersect (Core)/Config/Options.cs]: Fixed, dunno why that happened

File: Intersect.Client/Interface/Game/Bank/BankWindow.cs [hunk 2]
```diff
@@ -81,8 +81,18 @@ public void Update()
 
             X = mBankWindow.X;
             Y = mBankWindow.Y;
-            for (var i = 0; i < Globals.BankSlots; i++)
+            for (var i = 0; i < Options.Instance.BankOpts.MaxBankSize; i++)
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect.Client/Interface/Game/Bank/BankWindow.cs]: Good idea, done.

File: Intersect.Server/Database/PlayerData/Players/Guild.cs [hunk 3]
```diff
@@ -629,7 +629,7 @@ public void BankSlotUpdated(int slot)
         /// <param name="count"></param>
         public void ExpandBankSlots(int count)
         {
-            if (count > BankSlotsCount)
+            if (count > BankSlotsCount && count <= Options.Instance.BankOpts.MaxBankSize)
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect.Server/Database/PlayerData/Players/Guild.cs]: done

File: Intersect (Core)/Config/PlayerOptions.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public class PlayerOptions
         /// <summary>
         /// Number of bank slots a player has.
         /// </summary>
-        public int MaxBank = 100;
+        public int DefaultBankSize = 100;
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/Config/PlayerOptions.cs]: Done

File: Intersect (Core)/Config/Options.cs [hunk 5]
```diff
@@ -104,7 +106,7 @@ public class Options
 
         public static int MaxPlayerSkills => Instance.PlayerOpts.MaxSpells;
 
-        public static int MaxBankSlots => Instance.PlayerOpts.MaxBank;
+        public static int MaxBankSlots => Instance.PlayerOpts.DefaultBankSize;
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/Config/Options.cs]: Removed static, replaced existing usages

File: Intersect (Core)/Config/Options.cs [hunk 6]
```diff
@@ -77,6 +77,8 @@ public class Options
         public GuildOptions Guild = new GuildOptions();
 
         public LoggingOptions Logging = new LoggingOptions();
+        
+        public BankOptions BankOpts = new BankOptions();
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/Config/Options.cs]: done

File: Intersect (Core)/Config/Guilds/GuildOptions.cs [hunk 7]
```diff
@@ -32,7 +32,7 @@ public class GuildOptions
         /// <summary>
         /// Default number of storage slots in guild banks
         /// </summary>
-        public int GuildBankSlots { get; set; } = 50;
+        public int DefaultGuildBankSlots { get; set; } = 50;
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/Config/Guilds/GuildOptions.cs]: done

File: Intersect (Core)/Config/BankOptions.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace Intersect.Config
+{
+    public class BankOptions
+    {
+        /// <summary>
+        /// The max amount of bank slots possible, player, guild, or otherwise.
+        /// </summary>
+        public int MaxBankSize = 500;
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/Config/BankOptions.cs]: done

File: Intersect (Core)/Config/Options.cs (hunk 1 above)
Comment (Arufonsu) [reply Intersect (Core)/Config/Options.cs]: yeah, has happened to me as well 😆 👍🏽

**Review by lodicolo — APPROVED** [review]

**Pull request merged**